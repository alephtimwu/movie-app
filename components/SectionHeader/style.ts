import styled, { css } from '@emotion/native';
import { StyleSheet, View, Text } from 'react-native';

export const StyledHeading = styled.Text`
  text-align: left;
  font-family: Montserrat;
  color: #fff;
  font-size: 21px;
  line-height: 21px;
`;

export const StyledViewAll = styled.Text`
  text-align: right;
  font-family: Montserrat;
  color: #9da3b4;
  font-size: 10px;
  line-height: 10px;
`;
