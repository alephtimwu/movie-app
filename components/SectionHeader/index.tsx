import React, { useEffect, useState } from 'react';
import { View, Text } from 'react-native';
import globalStyles from '../../constants/styles';
import { StyledHeading, StyledViewAll } from './style';

export default ({ children, moreText = 'View All' }) => {
  return (
    <View style={[globalStyles.Flex, globalStyles.FlexBottom]}>
      <View style={globalStyles.Flex1}>
        <StyledHeading>{children}</StyledHeading>
      </View>
      <View style={globalStyles.Flex1}>
        <StyledViewAll>{moreText}</StyledViewAll>
      </View>
    </View>
  );
};
