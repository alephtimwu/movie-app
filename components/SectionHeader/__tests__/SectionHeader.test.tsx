import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
// import { CrpAll } from '__mocks__';
import UseAppContext from '../../../store/index';
import SectionHeader from '../index';

describe('SectionHeader =>', () => {
  let tree = null;

  beforeEach(() => {
    tree = render(<SectionHeader>Test</SectionHeader>);
  });

  test('SectionHeader should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('SectionHeader should render correctly with custom more text', () => {
    const { getAllByText, rerender } = tree;
    rerender(<SectionHeader moreText="more">Test</SectionHeader>);
  });
});
