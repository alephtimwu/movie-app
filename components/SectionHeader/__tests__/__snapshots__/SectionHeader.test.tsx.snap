// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader => SectionHeader should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "alignItems": "flex-end",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Montserrat",
          "fontSize": 21,
          "lineHeight": 21,
          "textAlign": "left",
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9da3b4",
          "fontFamily": "Montserrat",
          "fontSize": 10,
          "lineHeight": 10,
          "textAlign": "right",
        }
      }
    >
      View All
    </Text>
  </View>
</View>
`;
